bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java

```diff
@@ -251,11 +249,11 @@ private static ImageData autoScaleImageData (Device device, final ImageData imag
 		original.dispose ();
 		ImageData result = resultImage.getImageData (DPIUtil.getDeviceZoom ());
 		resultImage.dispose ();
-		return result;
-	case NEAREST:
-	default:
-		return imageData.scaledTo (scaledWidth, scaledHeight);
+		yield result;
 	}
+	case NEAREST -> imageData.scaledTo (scaledWidth, scaledHeight);
+	default -> imageData.scaledTo (scaledWidth, scaledHeight);
```

Comment:
`default` should be sufficient t cover both cases

Reply:
addressed the changes